src/lib/models/ConfigModel.ts

```diff
@@ -176,31 +192,29 @@ abstract class YamlFormatter {
 	}
 
 	private toYamlEngineOverridesForEngine(engineName: string): string {
-		const engineConfigDescriptor = this.userContext.core.getEngineConfigDescription(engineName);
-		const userEngineConfig = this.userContext.core.getEngineConfig(engineName);
-		const defaultEngineConfig = this.defaultContext.core.getEngineConfig(engineName);
+		const engineConfigDescriptor: ConfigDescription = this.codeAnalyzer.getEngineConfigDescription(engineName);
+		const userEngineConfig: EngineConfig = this.codeAnalyzer.getEngineConfig(engineName);
 
 		let yamlCode: string = '\n' +
-			this.toYamlSectionHeadingComment(engineConfigDescriptor.overview!) + '\n' +
+			this.toYamlSectionHeadingComment(engineConfigDescriptor.overview) + '\n' +
```

Comment:
It's nice to be able to remove all these `!` now that the core side has its own ConfigDescription which makes these values required (because it always fills it in).